gprofiler/profilers/java.py

```diff
@@ -284,6 +288,13 @@ def _check_disk_requirements(self) -> None:
                 f" required (on path: {self._output_path_host!r}"
             )
 
+    def _check_async_profiler_not_loaded(self) -> None:
+        if not self._java_safemode:
+            return
+        for mmap in self.process.memory_maps():
+            if "libasyncProfiler.so" in mmap.path and mmap.path != self._libap_path_process:
```

Comment:
Over different gProfiler versions, async-profiler version may change (and we include the path in the filename to have the DSO re-loaded). I suggest you check for gProfiler's prefix? like `/tmp/gproifler_tmp/...`?